src/mongocxx/test/collection_mocked.cpp

```diff
@@ -466,4 +466,24 @@ TEST_CASE("Collection", "[collection]") {
         REQUIRE(bulk_operation_execute_called);
         REQUIRE(bulk_operation_destroy_called);
     }
+
+    SECTION("Rename Fails, name stays the same", "collection::rename") {
+      bool collection_rename_called  = false, collection_rename_error = false;
+      std::string old_name = (std::string)mongo_coll.name();
+
+      collection_rename->interpose([&](mongoc_collection_t *, const char*, const char*, bool, bson_error_t*) {
+        collection_rename_called = true;
+        return false;
```

Comment:
verify here that the expected name (in this case "bad_name") is getting passed in